File: projects/octopus/octopus-server/src/main/java/octopus/api/projects/ProjectManager.java
package octopus.api.projects;

import java.io.IOException;

import octopus.server.projectmanager.OctopusProjectManager;

public class ProjectManager {

	public void create(String projectName)
	{
		try {
			OctopusProjectManager.create(projectName);
		} catch (IOException e) {
			throw new RuntimeException("Error creating project");
		}
	}

	public boolean doesProjectExist(String projectName)
	{
		return OctopusProjectManager.doesProjectExist(projectName);
	}

	public void delete(String projectName)
	{
		try {
			OctopusProjectManager.delete(projectName);
		} catch (IOException e) {
			throw new RuntimeException("Error deleting project");
		}
	}

	public Iterable<String> listProjects()
	{
		return OctopusProjectManager.listProjects();
	}

	public OctopusProject getProjectByName(String name)
	{
		return OctopusProjectManager.getProjectByName(name);
	}

}

File: python/octopus-tools/octopus/server/project_manager.py
import base64
from octopus.server.server_command import ServerCommand

class ProjectManager(object):

    def __init__(self, server_host, server_port):
        self.command = ServerCommand(server_host, server_port)

    def create(self, project_name):
        return self.command.execute_get_command("/manageprojects/create/{}".format(project_name))

    def delete(self, project_name):
        return self.command.execute_get_command("/manageprojects/delete/{}".format(project_name))

    def list(self):
        return self.command.execute_get_command("/manageprojects/list")
